controllers/chat/newChatAnonymous.js

```diff
@@ -0,0 +1,149 @@
+          ? getStreamChannel.state.pinned_messages
+          : [],
+        members: getStreamChannel.state.members ? getStreamChannel.state.members : [],
+        membership: getStreamChannel.state.membership ? getStreamChannel.state.membership : null,
+        duration: getStreamChannel.state.duration ? getStreamChannel.state.duration : null,
+        better_channel_members_object: getStreamChannel.state.members,
+        better_channel_members: Object.values(getStreamChannel.state.members)
+      };
+    } else {
+      // if channel not exist then create new channel
+      const channel = client.channel('messaging', {
+        members
+      });
+      console.log('channel', channel);
+      const findOrCreateChannel = await channel.create();
+      console.log('findOrCreateChannel', findOrCreateChannel);
+      findOrCreateChannel.members = await Promise.all(
+        findOrCreateChannel.members.map(async (member) => {
+          if (member.role !== 'owner') return member;
+
+          const anonInfo = await ChatAnonUserInfo.findOne({
+            where: {
+              channel_id: findOrCreateChannel.channel.id,
+              my_anon_user_id: member.user_id
+            }
+          });
+
+          let anonUserInfo = {};
+
+          if (anonInfo) {
+            anonUserInfo = anonInfo;
+            updateChannelAnonUserInfo(findOrCreateChannel, anonUserInfo);
+            return {
+              ...member,
+              anon_user_info_color_code: anonInfo.anon_user_info_color_code,
+              anon_user_info_color_name: anonInfo.anon_user_info_color_name,
+              anon_user_info_emoji_code: anonInfo.anon_user_info_emoji_code,
+              anon_user_info_emoji_name: anonInfo.anon_user_info_emoji_name
+            };
+          }
+
+          const emoji = BetterSocialConstantListUtils.getRandomEmoji();
+          const color = BetterSocialConstantListUtils.getRandomColor();
+          anonUserInfo = {
+            anon_user_info_color_code: color.code,
+            anon_user_info_color_name: color.color,
+            anon_user_info_emoji_code: emoji.emoji,
+            anon_user_info_emoji_name: emoji.name
+          };
+
+          updateChannelAnonUserInfo(findOrCreateChannel, anonUserInfo);
+          await ChatAnonUserInfoFunction.createChatAnonUserInfo(
+            ChatAnonUserInfo,
+            findOrCreateChannel.channel.id,
+            req?.userId,
+            members[0],
+            anonUserInfo
+          );
+
+          channel.updatePartial({
+            set: {
+              channel_type: CHANNEL_TYPE.ANONYMOUS,
+              ...anonUserInfo
+            }
+          });
+
+          return {
+            ...member,
+            anon_user_info_color_code: color.code,
+            anon_user_info_color_name: color.color,
+            anon_user_info_emoji_code: emoji.emoji,
+            anon_user_info_emoji_name: emoji.name
+          };
+        })
+      );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the else block where a new channel is created, there is no error handling for the case when `channel.create()` or `ChatAnonUserInfoFunction.createChatAnonUserInfo()` fails. It would be better to wrap these in a try-catch block to handle any potential errors that may occur during the execution of these functions.

```diff
+  try {
+    const findOrCreateChannel = await channel.create();
+    ...
+    await ChatAnonUserInfoFunction.createChatAnonUserInfo(
+      ChatAnonUserInfo,
+      findOrCreateChannel.channel.id,
+      req?.userId,
+      members[0],
+      anonUserInfo
+    );
+  } catch (error) {
+    console.error('Error creating channel or chat anonymous user info:', error);
+    return ErrorResponse.e500(res, 'Error creating channel or chat anonymous user info');
+  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->